HTML Datalist Element New Entry (#1319)
Codecademy/docs: content/html/concepts/elements/terms/datalist/datalist.md

---
Title: '<datalist>'
Description: A description of HTML datalist element. 
Subjects: 
-'Computer Science'
-'Web Design'
-'Web Development'
Tags:
-'Elements'
-'Form'

> -'Id'

Reviewer: ```suggestion
  -'Id'
```

'Input'
-'Lists'
'Values'
CatalogContent: 
-'learn-html'
-'paths/front-end-engineer-career-path'
---
#Datalist

**<datalist>** is a HTML element that displays pre-definede values to a user while typing into an input box. The user retains the freedom to enter any value, only the datalist will provide autosuggestions. 

##Syntax

<datalist> syntax consists of nested <option> tags within the opening and closing bracket of the opening and closing <datalist> tag similar lists. Each option element represents a suggestion that will appear on a drop-down list when a user types in a input box. 

A <datalist> is linked to a an <input> element by matching the list attribute of the <input> element with the id attribute of the <datalist> like the following:

> ```pseudo

Reviewer: The syntax example should show how the \<datalist\> is constructed, like this:
```
<input list="input_list" >

<datalist id="input_list">
  <option value="option1" >
  <option value="option2" >
  <option value="option3" >
</datalist>
```

Add a input element and add a list element with a value

Add a datalist element. Make the id attribute eqeal to the value of the list attribute of the input element.

In between the datalist tags, add option elements of your choice. Value atrribute determines what will be autosuggested. 

```

##Example

> The following example showcases a datalist that autosuggests different highschool subjects. Notice how the input tag and datalist are linked by matching values for the list attribute and id attribute for the input tag and datalist tag respectively. 

Reviewer: ```suggestion
The following example demonstrates a datalist that autosuggests different high school subjects. Notice how the input tag and datalist are linked by matching values for the list attribute and id attribute for the input tag and datalist tag respectively. 
```

```HTML
<html>

>     <head>

Reviewer: ```suggestion
  <head>
```

>         <title>Learning About Datalists</title>

Reviewer: ```suggestion
    <title>Learning About Datalists</title>
```

>     <head>

Reviewer: ```suggestion
  </head>
```

>     <body>

Reviewer: ```suggestion
  <body>
```

>         <input list="subjects" name="school-subjects" id="highschool">

Reviewer: ```suggestion
    <input list="subjects" name="school-subjects" id="highschool">
```

>         <datalist id="subjects">

Reviewer: ```suggestion
    <datalist id="subjects">
```

>             <option value="English">

Reviewer: ```suggestion
      <option value="English">
```

>             <option value="Math">

Reviewer: ```suggestion
      <option value="Math">
```

>             <option value="History">

Reviewer: ```suggestion
      <option value="History">
```

>             <option value="Physics">

Reviewer: ```suggestion
      <option value="Physics">
```

>             <option value="Biology">

Reviewer: ```suggestion
      <option value="Biology">
```

>             <option value="Economics">

Reviewer: ```suggestion
      <option value="Economics">
```

>         <datalist>

Reviewer: ```suggestion
    </datalist>
```

>     <body>

Reviewer: ```suggestion
  </body>
```

</html>

> ```

Reviewer: Remove the line breaks at the end of the file.
